### PR TITLE
Closes #1928 and improves #1957

### LINKED ISSUE #1928: DateRangePicker default state/ values
- Currently picker isn't picking default values passed through `dateRange`, need to check why this is not accepting that.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- For issue #1928 it's fixing the missed default state.
- For #1957 I am attempting to find the cause of the bug and fix that.

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js [hunk 1]
```diff
@@ -27,24 +25,18 @@ const useInitQueries = ({ eventId }) => {
...
-	}
-
 	const datetimes = pathOr([], ['datetimes', 'nodes'], datetimesData);
 	const datetimeIn = datetimes ? datetimes.map(({ id }) => id) : [];
-
 	const toasterTicketsMessage = 'initializing event editor tickets';
-	const [
-		fetchRelatedTickets,
-		{ data: ticketsData, error: ticketError, loading: loadingTickets, called },
-	] = useLazyQuery(GET_TICKETS, {
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: From what I understand [from this Apollo post ](https://blog.apollographql.com/apollo-client-now-with-react-hooks-676d116eeae2)is intended to be used upon some user action, check `Lazy evaluation for queries` paragraph.
***
And based on this discussion: https://github.com/apollographql/apollo-feature-requests/issues/119
`skip` option seem more suitable for cases when query needs to be executed conditionally.
I might be missing something, please correct me, if so.

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js [hunk 2]
```diff
@@ -27,24 +25,18 @@ const useInitQueries = ({ eventId }) => {
...
+	const skipTicketsQuery = datetimeIn.length === 0;
+
+	const { data: ticketsData, error: ticketError, loading: loadingTickets } = useQuery(GET_TICKETS, {
 		variables: {
 			first: 50,
 			where: {
 				datetimeIn,
 			},
+			skip: skipTicketsQuery,
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: On the other hand, passing `skip` true gives me this result:
image
@manzoorwanijk is something that needs to be done on serverside ?

Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: Also `skip` option is a nice declarative way of doing this chaining of queries.

**Comment by nicolad** [comment]
@tn3rb @manzoorwanijk 
in context of #1957:
image
***
I see that we have the data we need in `useInitQueries` available in DOM JSON dump:
image

Maybe should I go and do Apollo rehydration and query from there, instead of making these HTTP requests.

File: assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ const DateRangePicker = ({ onFieldUpdate, range, setRange }) => {
 	const appliedProps = {
 		allowSingleDayRange: true,
 		contiguousCalendarMonths: false,
-		dateRange: range,
+		value: range,
```
Comment (nicolad) [inline assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js]: This is a 1-line fix for #1928, that's why I didn't created a separate PR just for that.
Here is the fix:

Peek 2019-12-06 14-19

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js (hunk 2 above)
Comment (manzoorwanijk) [inline assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: `skip` is not to be added to variables, it should be added to options in `useMutation`.

File: assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js (hunk 3 above)
Comment (tn3rb) [reply assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js]: HA!  so much for my 8 hour estimate :\

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js (hunk 2 above)
Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: :man_facepalming:

Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: thx :)

File: assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js (hunk 3 above)
Comment (nicolad) [reply assets/prototype/domain/shared/dateRangeInput/DateRangePicker.js]: I got misled before by the example set by blueprint: https://github.com/palantir/blueprint/blob/develop/packages/docs-app/src/examples/datetime-examples/dateRangePickerExample.tsx#L36

**Pull request merged**